src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java

```diff
@@ -511,8 +510,7 @@ private static boolean isEmptyBlock(DetailAST ast, int parentType, int match) {
         if (type == TokenTypes.RCURLY) {
             final DetailAST parent = ast.getParent();
             final DetailAST grandParent = ast.getParent().getParent();
-            result = parentType == TokenTypes.SLIST
```

Comment:
if main type is rcurly and first child of parent is again said rcurly, then parent has to be slist.